### PR TITLE
Temp memory CommandBuffers

### PR DESCRIPTION
**Pull request by joeldevahl**
# **Please read;**
## **PR Workflow for the Graphics repository:**
* **All PRs must be opened as draft initially**
* Reviewers can be added while the PR is still in draft
* The PR can be marked as “Ready for Review” once the reviewers have confirmed that **no more changes are needed**
* Tests will start automatically after the PR is marked as “Ready for Review”
* **Do not use [skip ci]** - this can break some of our tooling
* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

---
### Testing status
Describe what manual/automated tests were performed for this PR

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs [hunk 1]
```diff
@@ -388,12 +388,14 @@ static void RenderSingleCamera(ScriptableRenderContext context, CameraData camer
 
             cameraData.xr.EndCamera(cmd, cameraData);
             context.ExecuteCommandBuffer(cmd); // Sends to ScriptableRenderContext all the commands enqueued since cmd.Clear, i.e the "EndSample" command
-            CommandBufferPool.Release(cmd);
+            cmd.Clear();
 
             using (new ProfilingScope(cmd, Profiling.Pipeline.Context.submit))
```
Comment (eh-unity) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs:393]: I think the correct fix is to pass null into ProfilingScope. 
The point is to profile context.Submit(); which doesn't take cmd, so the **scope** in cmdbuf will be empty with this modification.

**Comment by phi-lira** [comment]
It would be nice to add a test to catch possible regression on this.

**Comment by phi-lira** [comment]
@joeldevahl @eh-unity any updates on this?

**Comment by joeldevahl** [comment]
This is just a draft PR, I have no real task regarding working on this or landing the C++ changes.
If it is something you think is needed we can discuss it, involve leads and get the work planned. Using temp memory is potentially unsafe and we need to verify behaviour in a few corner cases.

Also it needs to be verified this actually fixes perf issues on android.

I can close the PR if you don't want it to hang around open as a draft.

**Comment by phi-lira** [comment]
Closing due to conversation on slack.

**Pull request closed without merging**